File: Week_04/numIslands.go
func numIslands(grid [][]byte) int {
    res:=0
    var dfs func(i int,j int)
    dfs = func(i int,j int){
        if 0<=i && i<len(grid) && 0<=j && j<len(grid[0]) && grid[i][j]=='1'{
            grid[i][j]='0'
            dfs(i-1,j)
            dfs(i+1,j)
            dfs(i,j-1)
            dfs(i,j+1)
        }
    }
    for i:=0;i<len(grid);i++{
        for j:=0;j<len(grid[0]);j++{
            if grid[i][j]=='1'{
                dfs(i,j)
                res++
            }
        }
    }
    return res
}

File: Week_04/numIslands.py
class Solution(object):
    def numIslands(self, grid):
        """
        :type grid: List[List[str]]
        :rtype: int
        """
        res=0
        def dfs(i,j):
            if 0<=i<len(grid) and 0<=j<len(grid[0]) and grid[i][j]=='1':
                grid[i][j]='0'
                dfs(i-1,j)
                dfs(i+1,j)
                dfs(i,j-1)
                dfs(i,j+1)
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j]=='1':
                    dfs(i,j)
                    res+=1
        return res
